### ISSUE TITLE
[Android] onNotification not called on click on notification

### ISSUE THREAD

**Issue by avencat** [calcazar/react-native-push-notification-CE#7]
Hi!
First of all, thanks for maintaining this lib! I'm using it in my project but I have a weird issue on Android (I'm using FCM and I do get the onNotification called when I receive the notification).
Indeed, when I tap on a notification, whether the app is opened or not, the onNotification method is not being called and if the app is already open, it entirely reloads... Do you have any workaround?

**RN**: 0.55.3
**lib**: `master` branch
**phone**: Xiaomi Redmi Note 4X

**Comment by calcazar** [comment]
@avencat are you calling the appStart() function on your component will mount?

**Comment by avencat** [comment]
@calcazar yes, in the componentDidMount on the top level of my app! :)

**Comment by avencat** [comment]
https://github.com/zo0r/react-native-push-notification/issues/697#issuecomment-391766566

**Comment by avencat** [comment]
Seems like it's a problem with FCM implementation: https://github.com/zo0r/react-native-push-notification/issues/697#issuecomment-391766566

**Comment by avencat** [comment]
All right, I'll take a look on that on Monday, but fyi I have the notification in the Notification Center, the onNotification method is called when I receive the notification but not when I tap on the notification on the Notification Center...

**Comment by avencat** [comment]
@calcazar so the payload is as mixed (silent and noisy), I tried only noisy but still, the tap on the notification doesn't trigger the `onNotification`, the `onNotification` is only triggered when I receive a notification with the app running… :(

**Comment by calcazar** [comment]
@avencat that's bizarre.. do you think you can create a repo for me to take a look at it? I've tried what I can on my end to try and replicate the issue but haven't had any luck

**Comment by avencat** [comment]
@calcazar I can't create a repo right now but I'm still looking into this issue, I found a similar issue on another library, I'll try the solution out! https://github.com/invertase/react-native-firebase/issues/1027#issuecomment-390557446 The issue could be the SplashActivity regarding this comment…

**Comment by nSimonFR** [comment]
I too have the same problem but I don't think it is related to SplashScreen as I do not have one configured in my Android Project...

**Comment by sergey-kozel** [comment]
Indeed, this can happen due to splash activity and it is a quite common issue on Android. And it's quite easy to fix that. When you tap on your push notification the system sent new intent with notification payload to your activity and under the hood this library hooks that intent and retrieves data. But when you have the splash activity the intent is sent to it, not to your main activity. All that you need to do is forwarding the extras further:

```
public class SplashActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);        
        Intent intent = new Intent(this, MainActivity.class);  
        if (getIntent().getExtras() != null) {
            intent.putExtras(getIntent().getExtras());
        }
        startActivity(intent);
        finish();
    }
}

**Comment by mikehardy** [comment]
There are similar issues with splash screen and firebase, if your module makes assumptions about being a single Activity, it's actually easy to integrate splash screen in a single activity - working code here https://github.com/crazycodeboy/react-native-splash-screen/issues/289

**Comment by TaraSinghDanu** [comment]
> Indeed, this can happen due to splash activity and it is a quite common issue on Android. And it's quite easy to fix that. When you tap on your push notification the system sent new intent with notification payload to your activity and under the hood this library hooks that intent and retrieves data. But when you have the splash activity the intent is sent to it, not to your main activity. All that you need to do is forwarding the extras further:
> 
> ```
> public class SplashActivity extends AppCompatActivity {
>     @Override
>     protected void onCreate(Bundle savedInstanceState) {
>         super.onCreate(savedInstanceState);        
>         Intent intent = new Intent(this, MainActivity.class);  
>         if (getIntent().getExtras() != null) {
>             intent.putExtras(getIntent().getExtras());
>         }
>         startActivity(intent);
>         finish();
>     }
> }
> ```

Thank you very much @sergey-kozel ,This works for me.

**Comment by Ajmal0197** [comment]
The issue is due to splash activity
Edit the manifest like this and it gonna work:

        <activity
            android:name=".SplashActivity"
            android:theme="@style/SplashTheme"
            android:label="@string/app_name">
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
        </activity>
      <activity
        android:name=".MainActivity"
        android:label="@string/app_name"
        android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
        android:windowSoftInputMode="adjustResize" >
          <intent-filter>
          <action android:name="android.intent.action.MAIN" />
          <category android:name="android.intent.category.INFO" />
          </intent-filter>
        </activity>

**Comment by mikehardy** [comment]
Everyone should use react-native-bootsplash at this point. react-native-splash-screen is buggy and unmaintained

https://github.com/zoontek/react-native-bootsplash
